src/Kaleidoscope.h

```diff
@@ -45,6 +45,27 @@ struct EventKey {
 
 class Kaleidoscope_;
 
+// Code generated by muli-line pre-processor macros is hard to read after
+//    pre-processing.
+//
+// When you want to check pre-processed code, e.g. for debugging or 
+// to understand what's going on, do the following:
+//
+// 1) Comment the definition of the __NL__ macro below.
+// 2) Generate the preprocessed code (it will contain a lot of __NL__ definitions).
+// 3) Open the pre-processed code in your favorite editor.
+//    3.1) Replace all __NL__ with newline characters. 
+//       In vim the command would be ':%s/__NL__/\r/g'.
+//    3.2) Autocorrect code-indenting to improve readability. This is necessary
+//       as pre-processor macros remove all whitespaces at the beginning of lines.
+//       With vim, the command gg=G helps (just type the characters one after 
+//       the other).
+// 4) Don't forget to uncomment the definition of the __NL__ macro again
+//       else the code won't compile.
+
+#define __NN__
+#define __NL__
```

Comment:
This macro is replaced with an empty string when compiled normally and remains in pre-processed code when this line is commented (see instructions).